### PR TITLE
[ADAM-1520] Bump to bdg-formats 0.11.0.

### LINKED ISSUE #1520: Bump to bdg-formats 0.11.0
See https://github.com/bigdatagenomics/bdg-formats/milestone/6.

@heuermh I'm thinking that we complete https://github.com/bigdatagenomics/bdg-formats/issues/114 and then cut bdg-formats 0.11.0. This will allow us to tackle #1257 for 0.23.0. This would push the genotype refactor (https://github.com/bigdatagenomics/bdg-formats/pull/108) out to 0.24.0. How does this sound to you?

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1520.

WIP; depends on https://github.com/bigdatagenomics/bdg-formats/pull/136.

@heuermh I figured it'd be easier to finish this off than to explain on  https://github.com/bigdatagenomics/bdg-formats/pull/136 why said PR was necessary. ;)

### CODE REVIEW

**Comment by fnothaft** [comment]
Jenkins, test this please.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 1]
```diff
@@ -63,7 +63,17 @@ import org.bdgenomics.adam.util.{
...
 }
-import org.bdgenomics.formats.avro._
+import org.bdgenomics.formats.avro.{
+  AlignmentRecord,
+  Contig,
+  Feature,
+  Fragment,
+  Genotype,
+  NucleotideContigFragment,
+  RecordGroup => RecordGroupMetadata,
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Is this to minimize changes, or is there a collision with `RecordGroup`?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Collision.

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Everywhere that it doesn't collide, I've gone with `RecordGroupMetadata`.

**Comment by fnothaft** [comment]
@heuermh I think that this is good to go. I'll cut the bdg-formats release tomorrow unless you have objections.

**Comment by fnothaft** [comment]
Oops! Clearly not quite done. Let me sort out the unit test failures and report back.

**Comment by heuermh** [comment]
How about merge https://github.com/bigdatagenomics/bdg-formats/pull/140 and then cut the bdg-formats 0.11.0 release?  I can pick up the necessary changes to ADAM in #1505.

**Comment by fnothaft** [comment]
> How about merge bigdatagenomics/bdg-formats#140 and then cut the bdg-formats 0.11.0 release? I can pick up the necessary changes to ADAM in #1505.

SGTM! I've merged bigdatagenomics/bdg-formats#140 upstream and will prep and cut the release later today.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**